### DISCUSSION TITLE
Would it be possible to add support for STM8 to this core or create a new one?

### DISCUSSION THREAD

**Discussion by rtek1000** [stm32duino/Arduino_Core_STM32#1768] [Variants] [answered]
Hello,

I found a core (sduino) that works with the basic parts, but I2C for example I can't implement the reading, even following the example code to pull the registers provided by ST in the AN3281:
https://github.com/tenbaht/sduino

I found some boards with STM8 for sale, it might be interesting to have support for this uC.

The I2C part is not completely autonomous and needs the CPU to handle the operations, even the example code adds a "dead time" to wait for the register to be updated. (AN3281/Project/STM8S/STM8S_I2C_Master_poll/src/i2c_master_poll.c).

There is also example code with interrupt, but it is not implemented in this core sduino apart from I2C, others such as External Interrupt and some Timers already are, but the project seems abandoned, 2 years without updates.

I found this example code, that makes use of interrupt, for Netduino:
https://blog.mark-stevens.co.uk/2015/05/stm8s-i2c-master-devices/

**Comment by rtek1000** [comment]
https://github.com/tenbaht/sduino/issues/143

**Reply by fpistm** [reply]
Maybe this one: https://github.com/stm32duino/Arduino_Core_STM8

**Reply by rtek1000** [reply]
Yes, I saw this other one, but when trying to install it in the IDE it gave an error, it said it doesn't have support for Linux.

**Reply by fpistm** [reply]
> Yes, I saw this other one, but when trying to install it in the IDE it gave an error, it said it doesn't have support for Linux.

Right. Unfortunately only windows is supported. No STM8 toolchain supporting C++ is available for Linux or you can use a virtual box to run Windows.

**Reply by rtek1000** [reply, marked as answer]
I'm going to try the Wire library available in the sduino repository, a person informed me of the existence of this library that I had not found:

> Why not use the examples provided with the wire library port included in the sduino repo?
> They work.
Ref.: https://github.com/tenbaht/sduino/issues/143

https://github.com/tenbaht/sduino/tree/development/sduino/stm8/libraries/Wire
